controllers/control/k8ssandratask_controller.go

```diff
@@ -0,0 +1,375 @@
+//+kubebuilder:rbac:groups=control.k8ssandra.io,namespace="k8ssandra",resources=cassandratasks/status,verbs=get
+
+func (r *K8ssandraTaskReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("K8ssandraTask", req.NamespacedName)
+
+	logger.Info("Fetching task")
+	kTask := &api.K8ssandraTask{}
+	if err := r.Get(ctx, req.NamespacedName, kTask); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	kcKey := kTask.GetClusterKey()
+	kc, kcExists, err := r.getCluster(ctx, kcKey)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Set owner reference so that the task is deleted when the K8ssandraCluster is deleted.
+	if kTask.OwnerReferences == nil && kcExists {
+		logger.Info("Setting owner reference", "K8ssandraCluster", kcKey)
+		if err := controllerutil.SetControllerReference(kc, kTask, r.Scheme); err != nil {
+			return ctrl.Result{}, fmt.Errorf("setting owner reference on K8ssandraTask: %s", err)
+		}
+		if err := r.Update(ctx, kTask); err != nil {
+			return ctrl.Result{}, fmt.Errorf("updating K8ssandraTask to set owner reference: %s", err)
+		}
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	// Handle deletion
+	if kTask.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(kTask, k8ssandraTaskFinalizer) {
+			logger.Info("Adding finalizer")
+			controllerutil.AddFinalizer(kTask, k8ssandraTaskFinalizer)
+			if err := r.Update(ctx, kTask); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{Requeue: true}, nil
+		}
+	} else { // The task is being deleted
+		if controllerutil.ContainsFinalizer(kTask, k8ssandraTaskFinalizer) {
+			// First time we've noticed the deletion, clean up dependents and remove the finalizer
+			if err := r.deleteCassandraTasks(ctx, kTask, kc, logger); err != nil {
+				return ctrl.Result{}, err
+			}
+			logger.Info("Removing finalizer")
+			controllerutil.RemoveFinalizer(kTask, k8ssandraTaskFinalizer)
+			if err := r.Update(ctx, kTask); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Handle TTL
+	if kTask.Status.CompletionTime != nil {
+		timeNow := metav1.Now()
+		deletionTime := calculateDeletionTime(kTask)
+		if deletionTime.IsZero() { // No TTL set, we're done
+			return ctrl.Result{}, nil
+		}
+
+		if deletionTime.Before(timeNow.Time) {
+			logger.Info("Deleting K8ssandraTask due to expired TTL")
+			err := r.Delete(ctx, kTask)
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
+
+		// Reschedule for later deletion
+		nextRunTime := deletionTime.Sub(timeNow.Time)
+		logger.Info("Scheduling K8ssandraTask deletion", "Delay", nextRunTime)
+		return ctrl.Result{RequeueAfter: nextRunTime}, nil
+	}
+
+	// Create CassandraTasks and/or gather their status
+	if !kcExists {
+		return r.reportInvalidSpec(ctx, kTask, "Unknown K8ssandraCluster %s.%s", kcKey.Namespace, kcKey.Name)
+	}
+	patch := client.MergeFrom(kTask.DeepCopy())
```

Comment:
There's a mixture of optimistic lock deletion, updates and softer patches. Is there a reason for different update strategies?

Reply:
Yeah I can use `Update()` for the status as well. I've changed it.